[KOGITO-8305] - [KSW-Guides] Deploying on OpenShift Clusters (#346)
kiegroup/kogito-docs: serverlessworkflow/modules/ROOT/pages/cloud/common/_proc_deploy_sw_quarkus_cli.adoc

[[proc-deploy-sw-application-quarkus-cli]]
=== Deploying your workflow application using Quarkus CLI

.Prerequisites
* Workflow application is ready.

> * Quarkus CLI is installed. +

Reviewer: I think `+` is not necessary. 
```suggestion
* Quarkus CLI is installed. 
```

For more information, see link:{quarkus_cli_url}[Building Quarkus Apps with Quarkus command line interface (CLI)].

.Procedure

> . Add the Quarkus extensions to deploy the `knative` service

Reviewer: ```suggestion
. Add the Quarkus extensions to deploy the `knative` service.
```

----
quarkus.knative.name=hello-workflow <1>
quarkus.kubernetes.deployment-target=knative <2>
quarkus.kubernetes.deploy=true <3>
quarkus.container-image.push=true <4>
quarkus.container-image.group={default_namespace} <5>
quarkus.container-image.registry={k8s_registry} <6>
quarkus.container-image.tag=1.0-SNAPSHOT <7>
----

<1> Property to set the Knative service name
<2> Property to set the target deployment type
<3> Property to set whether or not deploy on an active {platform} cluster
<4> {container_push}
<5> Property to define which registry group/namespace the built image belongs to
<6> Property to define to which registry the built image will be pushed to
<7> Sets the built Container Image tag

[IMPORTANT]
====
This feature works with Quarkus 2.10.2.Final or later. For more information, see
link:{quarkus_k8s_deploy_url}[link].
====
--

.Build and Deploy your workflow application
[source,shell,subs="attributes+"]
----
quarkus build -DskipTests \
  -Dquarkus.container-image.push=true \
  -Dquarkus.container-image.registry={k8s_registry} \
  -Dquarkus.container-image.group={default_namespace} \
  -Dquarkus.container-image.tag=1.0-SNAPSHOT
----

> Note that the maven profile activated is named as `container`, which provides the needed system properties to build the

Reviewer: ```suggestion
NOTE: The activated maven profile is named as a `container`, which provides the needed system properties to build the target container image.
```

> NOTE: The activated maven profile is named as a `container`, which provides the needed system properties to build the target container image.
> target container image.

Reviewer 2: ```suggestion
NOTE: The activated maven profile is named as a `container`, which provides the needed system properties to build the target container image.
```

This sentence is confusing and I'm not getting what it means.

Author: It should be the same from the build-workflow-image-with-quarkus.adoc.

[[proc-deploy-sw-application-quarkus-cli]]
=== Deploying your workflow application using Quarkus CLI

.Prerequisites
* Workflow application is ready.

> * Quarkus CLI is installed. 
> For more information, see link:{quarkus_cli_url}[Building Quarkus Apps with Quarkus command line interface (CLI)].

Reviewer 3: ```suggestion
* Quarkus CLI is installed. +
  For more information, see link:{quarkus_cli_url}[Building Quarkus Apps with Quarkus command line interface (CLI)].
```
